bench/src/utils.rs

```diff
@@ -2,23 +2,68 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-License-Identifier: MIT
 
-use serde::Serialize;
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
 use std::{
   collections::HashMap,
+  fs,
+  io::{BufRead, BufReader},
   path::PathBuf,
   process::{Command, Output, Stdio},
+  u64,
 };
 
+#[derive(Default, Clone, Serialize, Deserialize, Debug)]
+pub struct BenchResult {
+  pub created_at: String,
+  pub sha1: String,
+  pub exec_time: HashMap<String, HashMap<String, f64>>,
+  pub binary_size: HashMap<String, u64>,
+  pub max_memory: HashMap<String, u64>,
+  pub thread_count: HashMap<String, u64>,
+  pub syscall_count: HashMap<String, u64>,
+  pub cargo_deps: HashMap<String, usize>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Serialize)]
+pub struct StraceOutput {
+  pub percent_time: f64,
+  pub seconds: f64,
+  pub usecs_per_call: Option<u64>,
+  pub calls: u64,
+  pub errors: u64,
+}
+
+pub fn get_target() -> &'static str {
+  #[cfg(target_os = "macos")]
+  return "x86_64-apple-darwin";
+  #[cfg(target_os = "linux")]
+  return "x86_64-unknown-linux-gnu";
+  #[cfg(target_os = "windows")]
+  return unimplemented!();
```

Comment:
Is this intended?

Reply:
yup gonna work later to bring benches to other platforms but limited as the CI require X server.